### DISCUSSION TITLE
Issue when adding mesh adaptivity to a phase field simulation with elasticity of precipitate growth in a polycrystal

### DISCUSSION THREAD

**Discussion by simopier** [idaholab/moose#18003] [Q&A Modules: General] [answered]
I am trying to model the growth of precipitates in a polycrystal material with a phase field model that includes elasticity. Everything works without mesh adaptivity. However, when I try to use mesh adaptivity, I get huge errors in local displacements, as shown in the figure below: 

| Without mesh adaptivity      | With mesh adaptivity - the displacements are unphysical      |
|------------|-------------|
| <img src="https://user-images.githubusercontent.com/17445651/120701589-40f34c80-c470-11eb-839a-bf6332f5dd4b.png" width="250"> | <img src="https://user-images.githubusercontent.com/17445651/120701657-58323a00-c470-11eb-8bbb-465749a1393d.png" width="250"> |

There are a lot of things going on in these simulations, so I broke it down into features:
1. The material being a single crystal or a polycrystal.
2. The eigenstrain being null or non-null.
3. The simulations using mesh adaptivity of not

I have created simple input files (attached to this message) to isolate the features and identify which feature causes the problem. Everything in these input files remain the same, except for the specific features being listed here. They run with a combination of the `phase field` and `tensor mechanics` modules (`combined-opt` works well). The results are described in the following table:

| Input file name                        | Polycrystal  matrix  ?   |  Non-null eigenstrain  ?   | Mesh adaptivity ?     |  Runs properly ? |
|-----------------------------------------------------|-------------|-------------|-------------|-------------|
| Single_Elasticity_BasicEigenstrain_MeshAdaptivity.i   | NO               | yes               | yes                | yes                |
| Poly_Elasticity_0Eigenstrain_MeshAdaptivity.i             | yes               | NO                | yes                | yes                | 
| Poly_Elasticity_BasicEigenstrain_NoMeshAdaptivity.i | yes                |  yes              |   NO              | yes                |
| Poly_Elasticity_BasicEigenstrain_MeshAdaptivity.i      | yes                |  yes              | yes               | NO                |

As the results in the table show, it is only when all of these features are included together that the issue arrises. Surprisingly, the simulation with unphysical displacements does not show any an error, so I am not sure where the issue is coming from.

Any idea on what could be causing that problem? Or any tips on how I could figure out where the issue is coming from? Let me know if you need more information!

Thank you!

Edit: Here are the input files, along with the `Euler_angles_3G_same.txt` and `grain_growth_2D_voronoi_out.e-s020` files for the polycrystal
[MOOSE_discussion_input_file.zip](https://github.com/idaholab/moose/files/6639919/MOOSE_discussion_input_file.zip)

**Comment by jiangwen84** [comment, marked as answer]
@simopier Can you also upload `Euler_angles_3G_same.txt` and `grain_growth_2D_voronoi_out.e-s020` ?

**Reply by jiangwen84** [reply]
I think the issue might be related to `SolutionUserObject`. When you use `SolutionUserObject`, the solution vector can be read correctly, but the adaptivity information (e.g. hanging nodes constraints) will not be read. So your finite element mesh and/or solution will have some discontinuities. 

Another issue is to use adaptivity with stateful properties. Since you are using small deformation, you can set `incremental = false` to not use the stateful property.

**Reply by simopier** [reply]
Thanks for taking a stab at this!

> @simopier Can you also upload Euler_angles_3G_same.txt and grain_growth_2D_voronoi_out.e-s020 ?

My bad! I edited the question to include these two files. 

I'll try with `incremental = false` to see how it does, and come back to you.

Thank you!

**Reply by simopier** [reply]
I have tried to run the simulations with `incremental = false`, but it behaves the same way. The complete simulation (poly + non-null eigenstrain + mesh adaptivity) still has unphysical displacements. 

As for the potential issue with `SolutionUserObject`, I only use it at the very beginning of the simulation to establish the polycrystal (which does not evolve during the simulation), whereas the issues arrive only after the first time step. Moreover, I am not sure of what I could do instead of using the `SolutionUserObject`...

**Reply by GiudGiud** [reply]
You could use another restart option. Like using an exodus file.
This seems like a bug judging by the figure. Could you please create an issue for it?

**Reply by GiudGiud** [reply]
final ping before I mark this as answered
